packages/strapi-database/lib/validation/check-reserved-names.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const _ = require('lodash');
+const constants = require('../constants');
+
+class ModelError extends Error {
+  constructor(message) {
+    super(message);
+    this.stack = null;
+  }
+}
+
+const checkReservedAttributeNames = (model, { manager }) => {
+  const reservedNames = [...constants.RESERVED_ATTRIBUTE_NAMES];
+
+  if (_.has(model, 'options.timestamps')) {
+    const [connectorCreatedAt, connectorUpdatedAt] = manager.connectors.getByConnection(
+      model.connection
+    ).defaultTimestamps;
+
+    if (Array.isArray(model.options.timestamps)) {
+      const [
+        createdAtAttribute = connectorCreatedAt,
+        updatedAtAttribute = connectorUpdatedAt,
+      ] = model.options.timestamps;
+
+      reservedNames.push(createdAtAttribute, updatedAtAttribute);
+    } else {
+      reservedNames.push(connectorCreatedAt, connectorUpdatedAt);
+    }
+  }
+
+  const usedReservedAttributeNames = _.intersection(Object.keys(model.attributes), reservedNames);
+
+  if (usedReservedAttributeNames.length > 0) {
+    throw new ModelError(
+      `Model "${
+        model.modelName
+      }" is using reserved attribute names "${usedReservedAttributeNames.join(
+        ', '
+      )}".\n-> Make sure you are not using a reserved name or overriding the defined timestamp attributes.`
+    );
+  }
+};
+
+const checkReservedModelName = model => {
+  if (constants.RESERVED_MODEL_NAMES.includes(model.modelName)) {
```

Comment:
What about something like:
``` javascript
const checkReservedModelName = (model, { manager }) => {
  if (manager.getReservedNames().models.includes(model.modelName)) {
```

Reply:
I would rather keep the constant as the only source of truth in the module and leave the getReservedNames as an externalAPI

Reply:
It makes sense, but maybe we could clarify it somehow so it's clear for anyone reading the code?